htdocs/compta/stats/cabyprodserv.php

```diff
@@ -232,7 +232,7 @@
 	if ($selected_cat === -2) {	// Without any category
 		$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."categorie_product as cp ON p.rowid = cp.fk_product";
 	} elseif ($selected_cat) { 	// Into a specific category
-		$sql .= ", ".MAIN_DB_PREFIX."categorie as c, ".MAIN_DB_PREFIX."categorie_product as cp";
+		$sql .= " INNER JOIN ".MAIN_DB_PREFIX."categorie_product as cp ON (cp.fk_product = p.rowid) INNER JOIN ".MAIN_DB_PREFIX."categorie as c ON (cp.fk_categorie = c.rowid)";
```

Comment:
I think we should completely removed this because this join can generate duplicates record that will be counted twice into the SUM and later just keep the test line 266
```
$sql .= " AND (p.rowid IN ";
		$sql .= " (SELECT fk_product FROM ".MAIN_DB_PREFIX."categorie_product cp WHERE ";
		if ($subcat) $sql .= "cp.fk_categorie IN (".$listofcatsql.")";
		else $sql .= "cp.fk_categorie = ".$selected_cat;
		$sql .= "))";
```
So even if product is in several category when we search with $subcal set, we found and return product/line only once.

Reply:
good idea i do that